mp/src/game/shared/momentum/weapon/weapon_base.cpp

```diff
@@ -657,132 +701,190 @@ void CWeaponBase::DrawCrosshair()
+                }
 
-    if (!cl_crosshairusealpha.GetBool())
-    {
-        // Additive crosshair
-        vgui::surface()->DrawTexturedRect(iLeft, iHalfScreenHeight, iFarLeft, iHalfScreenHeight + iBarThickness);
-        vgui::surface()->DrawTexturedRect(iRight, iHalfScreenHeight, iFarRight, iHalfScreenHeight + iBarThickness);
+                vgui::surface()->DrawSetColor(crossColor); // could add cvar for dot color/alpha
+                vgui::surface()->DrawFilledRect(iHalfScreenWidth - 1, iHalfScreenHeight - 1, iHalfScreenWidth + 1, iHalfScreenHeight + 1); // could add cvar for dot size
+            }
+        }
     }
     else
     {
-        // Alpha-blended crosshair
-        vgui::surface()->DrawFilledRect(iLeft, iHalfScreenHeight, iFarLeft, iHalfScreenHeight + iBarThickness);
-        vgui::surface()->DrawFilledRect(iRight, iHalfScreenHeight, iFarRight, iHalfScreenHeight + iBarThickness);
-    }
+        //vgui::surface()->DrawSetColor(crossColor); // custom vtfs seem to only be opaque or clear/gone, likely due to DrawTexturedRect vs DrawFilledRect
 
-    int iTop = iHalfScreenHeight - (iCrosshairDistance + iBarSize);
-    int iBottom = iHalfScreenHeight + iCrosshairDistance + iBarThickness;
-    int iFarTop = iTop + iBarSize;
-    int iFarBottom = iBottom + iBarSize;
+        CHudTexture *pTexture = nullptr;
 
-    if (!cl_crosshairusealpha.GetBool())
-    {
-        // Additive crosshair
-        vgui::surface()->DrawTexturedRect(iHalfScreenWidth, iTop, iHalfScreenWidth + iBarThickness, iFarTop);
-        vgui::surface()->DrawTexturedRect(iHalfScreenWidth, iBottom, iHalfScreenWidth + iBarThickness, iFarBottom);
-    }
-    else
-    {
-        // Alpha-blended crosshair
-        vgui::surface()->DrawFilledRect(iHalfScreenWidth, iTop, iHalfScreenWidth + iBarThickness, iFarTop);
-        vgui::surface()->DrawFilledRect(iHalfScreenWidth, iBottom, iHalfScreenWidth + iBarThickness, iFarBottom);
+        if (strcmp(cl_crosshair_file.GetString(), "") == 0 || strcmp(cl_crosshair_file.GetString(), "null") == 0)
+            pTexture = gHUD.GetIcon("whiteAdditive"); //should probably just leave it to missing texture
+        else
+            pTexture = gHUD.GetIcon(cl_crosshair_file.GetString());
+
+        if (pTexture)
+            m_iCrosshairTextureID = pTexture->textureId;
+
+        vgui::surface()->DrawSetTexture(m_iCrosshairTextureID);
+
+        // make sure dynamic behaviour is ok
+        int iLeft = iHalfScreenWidth - (iBarSize + iCrosshairDistance) / 2;
+        int iTop = iHalfScreenHeight - (iBarSize + iCrosshairDistance) / 2;
+        int iRight = iHalfScreenWidth + (iBarSize + iCrosshairDistance) / 2 + (iBarSize + iCrosshairDistance) % 2;
+        int iBottom = iHalfScreenHeight + (iBarSize + iCrosshairDistance) / 2 + (iBarSize + iCrosshairDistance) % 2;
+
+        vgui::surface()->DrawTexturedRect(iLeft, iTop, iRight, iBottom);
     }
+	*/
```

Comment:
Very nice. Seems like the Linux builder is upset but also, you're in version control, there would be no need to commit commented out code, since if you ever need it back you can just go back to the previous commit that had it. You can remove this commented code.